frontend/src/forage/components/content/customMarkerMapComponent/CustomMarkerMapComponent.tsx

```diff
@@ -0,0 +1,46 @@
+import L from 'leaflet';
+import {CustomMarkerMapComponentProps} from "./CustomMarkerMapComponentProps.ts";
+import {Marker, Popup} from "react-leaflet";
+import {useEffect, useState} from "react";
+
+const CustomMarkerMapComponent = (props: Readonly<CustomMarkerMapComponentProps>) => {
+
+    const [customIcon, setCustomIcon] = useState<L.Icon>(
```

Comment:
You might need to wait with filling your useState until the props have arrived from the parent component. I suggest initializing the state as undefined and using a useEffect that calls the setter of the state as soon as the props have arrived

Reply:
or you could also just write `if (!props) return <p> Loading..</p>` somewhere at the top to prevent react from trying to render something that is not there